vsintegration/src/FSharp.Editor/Common/RoslynHelpers.fs

```diff
@@ -172,9 +172,15 @@ module internal OpenDeclarationHelper =
         let mutable minPos = None
 
         let insert line lineStr (sourceText: SourceText) : SourceText =
-            let pos = sourceText.Lines.[line].Start
+            let ln = sourceText.Lines.[line]
+            let pos = ln.Start
             minPos <- match minPos with None -> Some pos | Some oldPos -> Some (min oldPos pos)
-            sourceText.WithChanges(TextChange(TextSpan(pos, 0), lineStr + Environment.NewLine))
+
+            // find the line break characters on the previous line to use, Environment.NewLine should not be used
+            // as it makes assumptions on the line endings in the source.
+            let lineBreak = ln.Text.ToString(TextSpan(ln.End, ln.EndIncludingLineBreak - ln.End))
```

Comment:
What if we're the last line?

Reply:
In theory, if it were the last line without a line break, then `lineBreak` would be empty. With my (limited) understanding of the callers of `lineBreak`, I think the line in question is referring to the last line in the open declarations. In practice, I don't know if that would ever be the last line in the source file.

I did try to put a fake open declaration at the bottom of the source file, and even with the fix it still puts the line in the first open decl block above the usage (which makes sense to me).

I did look a bit further, and I think outside of this change there is another issue here, as this code inserts the line after the last open declaration it could cause two open decls to run together on the same line, since the first one may not have a line ending itself, e.g.: `open System` -> `open Systemopen System.IO`. I believe my change only alters whether the trailing line ending would be present - even in this case I think my change keeps it consistent (no line break).